compiler/test/tests/cobalt/Parser/AExprParserTest.hs

```diff
@@ -6,29 +6,113 @@ import Text.Megaparsec
-             Left  e -> error $ show e
-             Right x -> x)
---}
-
-testAExprParserInt :: Test
-testAExprParserInt = do
-    let code1 = "1000"
-    let test1 = TestCase $ assertEqual code1
-                    (IntConst 1000)
-                    (case (parse (aExpr) "" code1) of
-                         Left  e -> error $ show e
-                         Right x -> x)
-    let code2 = "-1000"
-    let test2 = TestCase $ assertEqual code2
-                    (Neg (IntConst 1000))
-                    (case (parse (aExpr) "" code2) of
-                         Left  e -> error $ show e
-                         Right x -> x)
-    TestList [test1, test2]
\ No newline at end of file
+
+import AST.AST
+import Parser.ExprParser
+
+testAExprParser :: Test
+testAExprParser = do
+    let codeAdd = "x + 100"
+    let testAdd = TestCase $ assertEqual codeAdd
+                       (ABinary Add (Var "x") (IntConst 100))
+                       (case (parse aExpr "" codeAdd) of
+                           Left  e -> error $ show e
+                           Right x -> x)
+
+    let codeSubtract = "x - 100"
+    let testSubtract = TestCase $ assertEqual codeSubtract
+                       (ABinary Subtract (Var "x") (IntConst 100))
+                       (case (parse aExpr "" codeSubtract) of
+                           Left  e -> error $ show e
+                           Right x -> x)
+
+    let codeMultiply = "x * 100"
+    let testMultiply = TestCase $ assertEqual codeMultiply
+                       (ABinary Multiply (Var "x") (IntConst 100))
+                       (case (parse aExpr "" codeMultiply) of
+                           Left  e -> error $ show e
+                           Right x -> x)
+
+    let codeDivide = "x / 100"
+    let testDivide = TestCase $ assertEqual codeDivide
+                       (ABinary Divide (Var "x") (IntConst 100))
+                       (case (parse aExpr "" codeDivide) of
+                           Left  e -> error $ show e
+                           Right x -> x)
+
+    let codeMixed = "x / 100 * y + 200 - z"
+    let testMixed = TestCase $ assertEqual codeMixed
+                       (ABinary Subtract (ABinary Add (ABinary Multiply (ABinary Divide (Var "x") (IntConst 100)) (Var "y")) (IntConst 200)) (Var "z"))
+                       (case (parse aExpr "" codeMixed) of
+                           Left  e -> error $ show e
+                           Right x -> x)
+
+    let codeNegative = "- 100"
+    let testNegative = TestCase $ assertEqual codeNegative
+                       (Neg (IntConst 100))
+                       (case (parse aExpr "" codeNegative) of
+                           Left  e -> error $ show e
+                           Right x -> x)
+
+    TestList [ testAdd
+             , testSubtract
+             , testMultiply
+             , testDivide
+             , testMixed
+             , testNegative
+             ]
+
+{-
+testAExprParserExpr :: Test
+testAExprParserExpr = do
```

Comment:
This commented block is basically a duplication of previous tests.

Reply:
The commented out tests here highlight an issue I raised. They should pass but until we fix #458 they won't.